system_test.sh

```diff
@@ -22,7 +22,19 @@ function try { let tests+=1; this="$1"; }
 
 function assert {
 	[[ "$1" == "$2" ]] && { printf "."; return; }
-	printf "\nFAIL: $this\n'$1' != '$2'\n"; exit 1
+	printf "\nFAIL: $this\n'$1' != '$2'\n"
+
+	echo "*************************************"
+	echo "All files in temporary directory and and a peek of their contents:"
+	head -n 50 $(find -type f)
```

Comment:
This needs to be `$(find $tmp -type f)` or `$(find . -type f)` to be compatible with BSD find(1)

Reply:
updated